### PR TITLE
Introduce overloaded constructor on AzureEventSourceListener

### PR DESCRIPTION
**Pull request by pharring**
Following on from #43947, this PR introduces an overloaded constructor for `AzureEventSourceListener` that takes a (simpler) delegate without the formatted message. The intent is to give callers a way to avoid or defer the cost of formatting the message. 

This will be a useful performance improvement for some "always on" listeners such as the AzureEventSourceLogForwarder in Microsoft.Extensions.Azure and Azure.Monitor.OpenTelemetry.

### CODE REVIEW

**Comment by pharring** [comment]
@christothes I presume this change has to be merged into Azure.Core before we can update the other SDK libraries (extensions and monitor) to consume it. If there's already a process to coordinate changes like this across libraries, please LMK.

**Comment by jsquire** [comment]
@pharring :  Thanks very much for your contribution and continued help improving the Azure SDK developer experience!

I agree that this change would be helpful for scenarios when the Extensions [AzureEventSourceLogForwarder](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/extensions/Microsoft.Extensions.Azure/src/AzureEventSourceLogForwarder.cs) or Monitor [AzureEventSourceLogForwarder](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Internals/AzureSdkCompat/AzureEventSourceLogForwarder.cs) are used, as they ignore the formatted message string in the listener callback and delegate formatting to `EventSourceEvent`.   

Because this is a change to the public API, we'll need approval from the .NET architects to move forward.   @KrzysztofCwalina and @tg-msft :  Would you please offer your thoughts?

**Comment by jsquire** [comment]
> @christothes I presume this change has to be merged into Azure.Core before we can update the other SDK libraries (extensions and monitor) to consume it. If there's already a process to coordinate changes like this across libraries, please LMK.

There's no formal process.   You're correct in that this would need to be merged and then would either need to wait for a Core release or temporarily shift the Monitor and Extensions source to a project reference temporarily.  Either way, we would generally use GitHub issues to track the individual changes needed.    In this case, since we have no urgency, I'd prefer to use the first approach and just catch the August release of Azure Core that should drop in two weeks.

File: sdk/core/Azure.Core/tests/AzureEventSourceTests.cs
Comment (KrzysztofCwalina) [inline sdk/core/Azure.Core/tests/AzureEventSourceTests.cs:42]: can we keep a test verifying that we do the formatting when the other constructor overload is used?

Comment (pharring) [reply sdk/core/Azure.Core/tests/AzureEventSourceTests.cs:42]: There are several in the existing file "AzureEventSourceListenerTests.cs" (not touched in this PR)

**Comment by pharring** [comment]
@KrzysztofCwalina: waiting for your approval on the public API change.

**Comment by KrzysztofCwalina** [comment]
APIs look good to me.

**Comment by pharring** [comment]
@jsquire I'm fairly certain that the test failures aren't related to this PR. Can this be merged now, or do we need to get clean?

**Comment by jsquire** [comment]
@pharring : I agree that this is unrelated.  It's a timeout impacting macOS for another area.   I've kicked just that leg to see if we can get it to play nice and, if not, I'll force the merge.

**Review by jsquire — APPROVED** [review]

**Pull request merged**
